### PR TITLE
expose keeporder for dataset in training

### PR DESCRIPTION
**Pull request by dding3**
For some senarios, use need keeping original data order for training, eg. gcn network. However our current implementation will shuffle the data and doesn't keep the original order. The PR is used to support user specfiy whether need keep the training data's original order.

### CODE REVIEW

**Comment by jason-dai** [comment]
Please provide a description of the PR (e.g., what problem does it solve?)

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 1]
```diff
@@ -438,10 +440,14 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
       nbEpoch: Int = 10,
       validationData: RDD[Sample[T]] = null,
       featurePaddingParam: PaddingParam[T] = null,
-      labelPaddingParam: PaddingParam[T] = null)(implicit ev: TensorNumeric[T]): Unit = {
+      labelPaddingParam: PaddingParam[T] = null,
+      keepOrder: Boolean = false,
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:444]: change to `shuffleData`

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala [hunk 2]
```diff
@@ -368,15 +368,16 @@ object DataSet {
    * @tparam T
    * @return
    */
-  def rdd[T: ClassTag](data: RDD[T], partitionNum: Int = Engine.nodeNumber()
+  def rdd[T: ClassTag](data: RDD[T], partitionNum: Int = Engine.nodeNumber(),
+    keepOrder: Boolean = false, groupSize: Int = 1
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala:372]: change to `shuffleData`

**Comment by dding3** [comment]
http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-Dllib-Validation/269/. The failure is not related to this PR, merge the pr first.

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 3]
```diff
@@ -433,15 +435,19 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
...
-      validationData: RDD[Sample[T]] = null,
-      featurePaddingParam: PaddingParam[T] = null,
-      labelPaddingParam: PaddingParam[T] = null)(implicit ev: TensorNumeric[T]): Unit = {
+           x: RDD[Sample[T]],
+           batchSize: Int = 32,
+           nbEpoch: Int = 10,
+           validationData: RDD[Sample[T]] = null,
+           featurePaddingParam: PaddingParam[T] = null,
+           labelPaddingParam: PaddingParam[T] = null,
+           shuffleData: Boolean = false,
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:444]: should default to true

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala [hunk 4]
```diff
@@ -368,15 +368,16 @@ object DataSet {
    * @tparam T
    * @return
    */
-  def rdd[T: ClassTag](data: RDD[T], partitionNum: Int = Engine.nodeNumber()
+  def rdd[T: ClassTag](data: RDD[T], partitionNum: Int = Engine.nodeNumber(),
+    shuffleData: Boolean = false, groupSize: Int = 1
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala:372]: should default to true

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 5]
```diff
@@ -315,16 +315,18 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
    */
   private def toDataSet(x: RDD[Sample[T]], batchSize: Int,
     featurePaddingParam: PaddingParam[T] = null,
-    labelPaddingParam: PaddingParam[T] = null): DataSet[MiniBatch[T]] = {
+    labelPaddingParam: PaddingParam[T] = null,
+    shuffleData: Boolean = false,
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:319]: should default to true

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala [hunk 6]
```diff
@@ -368,15 +368,16 @@ object DataSet {
...
     ): DistributedDataSet[T] = {
     new CachedDistriDataSet[T](
       data.coalesce(partitionNum, true)
         .mapPartitions(iter => {
           Iterator.single(iter.toArray)
         }).setName("cached dataset")
         .cache()
-    )
+      , shuffleData, groupSize)
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/feature/dataset/DataSet.scala:380]: shuffleData == false
